emcmake.py

```diff
@@ -12,25 +12,47 @@
 from tools import utils
 from subprocess import CalledProcessError
 
+# Print message to stderr and exit early.
+def error(message, exit_code=1, file=sys.stderr):
+  print(message, file=file)
+  sys.exit(exit_code)
 
-#
-# Main run() function
-#
-def run():
-  if len(sys.argv) < 2 or sys.argv[1] in ('--version', '--help'):
-    print('''\
+# Conditionally add a toolchain and run emulator.
+def create_command():
+  if len(sys.argv) < 2 or sys.argv[1] in ('--help', '--version'):
+    error('''\
 emcmake is a helper for cmake, setting various environment
 variables so that emcc etc. are used. Typical usage:
 
   emcmake cmake [FLAGS]
-''', file=sys.stderr)
-    return 1
+''')
+
+  unsupported = {
+    '--build',
+    '--install',
+    '--open',
+    '-E',
+    '--find-package',
+    '--help',
+    '--version',
+  }
 
   args = sys.argv[1:]
+  if len(args) < 2 or args[1] in unsupported:
```

Comment:
Should that be `if len(args) > 1 and args[1] in unsupported:`?

If there is only one arg don't we still want to activate emcmake logic?

Reply:
This does handle `emcmake` logic, since the first argument will always be `emcmake.py`, the second argument will be `cmake`. So, we do the following:

1. Remove `emcmake.py` from the argument list  (`args = sys.argv[1:]`).
2. If the wrapper is called only with `cmake`, don't add the toolchain `if len(args) < 2`.
3. If the wrapper is called with an unsupported command, don't add the toolchain `args[1] in unsupported`.

We can still add the toolchain if the command is called as `emcmake cmake`, which would be the case when `len(args) == 1`, but in this case, the two will be identical.

Reply:
> If there is only one arg don't we still want to activate emcmake logic?

The rationale here is that invoking `cmake` without any arguments will only print a help text instead of process a CMakeLists.txt file. I.e.

```
$ cmake
Usage

  cmake [options] <path-to-source>
  cmake [options] <path-to-existing-build>
  cmake [options] -S <path-to-source> -B <path-to-build>

Specify a source directory to (re-)generate a build system for it in the
current working directory.  Specify an existing build directory to
re-generate its build system.

Run 'cmake --help' for more information.
```